zign/api.py

```diff
@@ -34,8 +109,69 @@ def __str__(self):
         return 'Configuration error: {}'.format(self.msg)
 
 
-def get_config():
-    return stups_cli.config.load_config('zign')
+class ClientRedirectHandler(tools.ClientRedirectHandler):
+    '''Handles OAuth 2.0 redirect and return a success page if the flow has completed.'''
+
+    def do_GET(self):
+        '''Handle the GET request from the redirect.
+
+        Parses the token from the query parameters and returns a success page if the flow has completed'''
+
+        self.send_response(200)
+        self.send_header('Content-type', 'text/html')
+        self.end_headers()
+        query_string = urlparse(self.path).query
+
+        if not query_string:
+            self.wfile.write(EXTRACT_TOKEN_PAGE.format(port=self.server.server_port).encode('utf-8'))
+        else:
+            self.server.query_params = parse_qs(query_string)
+            if 'access_token' in self.server.query_params:
+                page = SUCCESS_PAGE
+            else:
+                page = ERROR_PAGE
+            self.wfile.write(page.encode('utf-8'))
+
+
+def get_config(config_module=None, override=dict()):
```

Comment:
never use `dict()` as default parameter in Python --- it's only initialized once, so you are operating on a singleton!

Better: 

```
def foo(a=None):
  a = a or {} # use empty dict if a is None or empty